src/ansiblelint/rules/no_jinja_when.py

```diff
@@ -66,6 +69,40 @@ def matchtask(
     ) -> bool | str:
         return "when" in task.raw_task and not self._is_valid(task.raw_task["when"])
 
+    def transform(
+        self,
+        match: MatchError,
+        lintable: Lintable,
+        data: CommentedMap | CommentedSeq | str,
+    ) -> None:
+        if match.tag == self.id:
+            task = self.seek(match.yaml_path, data)
+            key_to_check = ("when", "changed_when", "failed_when")
+            for _ in range(len(task)):
+                k, v = task.popitem(False)
+                if k == "roles" and isinstance(v, list):
+                    transform_for_roles(v, key_to_check=key_to_check)
+                elif k in key_to_check:
+                    processed_value = re.sub(r"{{ (.*?) }}", r"\1", v)
```

Comment:
@shatakshiiii we can directly assign this value to `v` instead of assigning on the next line